packages/core/src/diff/changes/directive-usage.ts

```diff
@@ -0,0 +1,206 @@
+  path: string,
+): Change<ChangeType.DirectiveUsageAdded> {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+    },
+    message: `Directive '${directive.name.value}' was added`,
+    meta: {},
+    path,
+    type: ChangeType.DirectiveUsageAdded,
+  };
+}
+
+export function directiveUsageRemoved(
+  directive: ConstDirectiveNode,
+  path: string,
+): Change<ChangeType.DirectiveUsageRemoved> {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+    },
+    message: `Directive '${directive.name.value}' was removed`,
+    meta: {},
+    path,
+    type: ChangeType.DirectiveUsageRemoved,
+  };
+}
+
+export function directiveUsageBuildUnionMemberAddedMessageFromMeta(
+  args: DirectiveUsageUnionMemberAddedChange,
+) {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+      reason:
+        'Adding a union member to a union can cause existing queries that use this union member in a fragment spread to error.',
+    },
+    type: ChangeType.DirectiveUsageUnionMemberAdded,
+    message: `Directive '${args.meta.addedUnionMemberTypeName}' was added to Union type '${args.meta.unionName}'`,
+    meta: args.meta,
+    path: args.meta.unionName,
+  } as const;
+}
+
+export function directiveUsageUnionMemberAdded(
+  union: GraphQLUnionType,
+  directive: ConstDirectiveNode,
+): Change<ChangeType.DirectiveUsageUnionMemberAdded> {
+  return directiveUsageBuildUnionMemberAddedMessageFromMeta({
+    type: ChangeType.DirectiveUsageUnionMemberAdded,
+    meta: {
+      unionName: union.name,
+      addedUnionMemberTypeName: directive.name.value,
+    },
+  });
+}
+
+export function directiveUsageBuildUnionMemberRemovedMessageFromMeta(
+  args: DirectiveUsageUnionMemberRemovedChange,
+) {
+  return {
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+      reason:
+        'Removing a union member from a union can cause existing queries that use this union member in a fragment spread to error.',
+    },
+    type: ChangeType.DirectiveUsageUnionMemberRemoved,
+    message: `Directive '${args.meta.removedUnionMemberTypeName}' was removed from Union type '${args.meta.unionName}'`,
+    meta: args.meta,
+    path: args.meta.unionName,
+  } as const;
+}
+
+export function directiveUsageUnionMemberRemoved(
+  union: GraphQLUnionType,
+  directive: ConstDirectiveNode,
+): Change<ChangeType.DirectiveUsageUnionMemberRemoved> {
+  return directiveUsageBuildUnionMemberRemovedMessageFromMeta({
+    type: ChangeType.DirectiveUsageUnionMemberRemoved,
+    meta: {
+      unionName: union.name,
+      removedUnionMemberTypeName: directive.name.value,
+    },
+  });
+}
+
+export function directiveUsageEnumAddedFromMeta(args: DirectiveUsageEnumAddedChange) {
+  return {
+    type: ChangeType.DirectiveUsageEnumAdded,
+    criticality: {
+      level: CriticalityLevel.Dangerous,
+      reason: `Directive '${args.meta.addedDirectiveName}' was added to Enum '${args.meta.enumName}'`,
```

Comment:
Reason should not be the same as the message, right? 
Also, `Enum` -> `enum` (same for all other keywords)